File: app/src/main/java/com/infy/dagger2implementation/database/datasource/LocationDataSource.java
package com.infy.dagger2implementation.database.datasource;

import com.infy.dagger2implementation.database.dao.LocationDao;
import com.infy.dagger2implementation.database.model.LocationData;
import com.infy.dagger2implementation.database.repository.LocationRepository;

import javax.inject.Inject;

public class LocationDataSource implements LocationRepository {
    private LocationDao locationDao;

    @Inject
    public LocationDataSource(LocationDao locationDao) {
        this.locationDao = locationDao;
    }

    @Override
    public long insert(LocationData locationData) {
        return locationDao.insert(locationData);
    }

    @Override
    public int delete(LocationData locationData) {
        return locationDao.delete(locationData);
    }

}

File: app/src/main/java/com/infy/dagger2kotlin/database/datasource/LocationDataSource.kt
package com.infy.dagger2kotlin.database.datasource

import com.infy.dagger2kotlin.database.dao.LocationDao
import com.infy.dagger2kotlin.database.model.LocationData
import com.infy.dagger2kotlin.database.repository.LocationRepository

import javax.inject.Inject

class LocationDataSource @Inject
constructor(private val locationDao: LocationDao) : LocationRepository {

    override fun insert(locationData: LocationData): Long {
        return locationDao.insert(locationData)
    }

    override fun delete(locationData: LocationData): Int {
        return locationDao.delete(locationData)
    }

}
